### ISSUE TITLE
Feathers Themes

### ISSUE THREAD

**Issue by alice-i-cecile** [bevyengine/bevy#23025]
## Goal Description

Themes are an important tool for both accessibility and self-expression. `feathers`, Bevy's first-party UI widget toolkit (akin to MaterialUI), should support them. Users should be able to globally configure (via code) common knobs like font choices and colors across all `feathers` widgets in a single place. We should ship at least one default themes out of the box.

## Goal Status

This is a potential [Bevy Project Goal](https://bevy.org/learn/contribute/project-information/project-goals/). Its state on the [Project Goals Board](https://github.com/orgs/bevyengine/projects/23/views/1) determines if it is Proposed, Postponed, Blocked (Approved), Inactive (Approved), Active (Approved), Completed, or Declined.

- **Staffing SMEs**: Not yet staffed
- **Working Group Link**: Working Group not yet formed
- **Design Documents**: Design documents not yet written

**Comment by alice-i-cecile** [comment]
Currently (0.19), we store a [`ThemeProps`](https://dev-docs.bevy.org/bevy/feathers/theme/struct.ThemeProps.html) inside of a [`UiTheme`](https://dev-docs.bevy.org/bevy/feathers/theme/struct.UiTheme.html).

**Comment by viridia** [comment]
Theming is a sequence of goals of exponentially increasing ambition:
* Ship a single theme (what we have now)
* Ship two themes (light and dark)
* Ship an example custom theme ("feathers but purple")
* Ship an algorithmic theme generator (give it a primary color and some params, and it produces a theme object)
* Ship an interactive theme editor.

**Comment by cart** [comment]
I'm considering this Goal "completed", as we already have themes as specified in the description. Followup work, such as light themes, theme editors, etc can be done later.
